tests/sync/test_accessibility.py

```diff
@@ -216,57 +215,95 @@ def test_accessibility_filtering_children_of_leaf_nodes_rich_text_editable_field
+                    {"role": "text", "name": "Edit this image:"},
+                    {"role": "img", "name": "my fake image"},
+                ],
+                "value": "Edit this image: ",
+            }
+        else:
+            golden = {
+                "role": "textbox",
+                "name": "",
+                "multiline": True,
+                "value": "Edit this image: ",
+                "children": [
+                    {"role": "text", "name": "Edit this image:"},
+                    {"role": "img", "name": "my fake image"},
+                ],
+                "value": "Edit this image: ",
+            }
     snapshot = page.accessibility.snapshot()
     assert snapshot["children"][0] == golden
 
 
 # Firefox does not support contenteditable="plaintext-only".
 # WebKit rich text accessibility is iffy
 @pytest.mark.only_browser("chromium")
-def test_accessibility_plain_text_field_with_role_should_not_have_children(page):
+def test_accessibility_plain_text_field_with_role_should_not_have_children(
+    page, browser_channel
+):
     page.set_content(
         """
     <div contenteditable="plaintext-only" role='textbox'>Edit this image:<img src="fakeimage.png" alt="my fake image"></div>"""
     )
     snapshot = page.accessibility.snapshot()
-    assert snapshot["children"][0] == {
-        "role": "textbox",
-        "name": "",
-        "value": "Edit this image:",
-    }
+    if browser_channel:
+        assert snapshot["children"][0] == {
+            "name": "",
+            "role": "textbox",
+            "value": "Edit this image:",
+        }
+    else:
+        assert snapshot["children"][0] == {
+            "multiline": True,
+            "name": "",
+            "role": "textbox",
+            "value": "Edit this image:",
+        }
 
 
 @pytest.mark.only_browser("chromium")
 def test_accessibility_plain_text_field_without_role_should_not_have_content(
-    page,
+    page, browser_channel
 ):
     page.set_content(
         """
     <div contenteditable="plaintext-only">Edit this image:<img src="fakeimage.png" alt="my fake image"></div>"""
     )
     snapshot = page.accessibility.snapshot()
-    assert snapshot["children"][0] == {"role": "generic", "name": ""}
+    if browser_channel:
+        assert snapshot["children"][0] == {"name": "", "role": "generic"}
+    else:
+        assert snapshot["children"][0] == {
+            "name": "",
+            "role": "generic",
+            "value": "Edit this image:",
+        }
 
 
 @pytest.mark.only_browser("chromium")
 def test_accessibility_plain_text_field_with_tabindex_and_without_role_should_not_have_content(
-    page,
+    page, browser_channel
 ):
     page.set_content(
         """
     <div contenteditable="plaintext-only" tabIndex=0>Edit this image:<img src="fakeimage.png" alt="my fake image"></div>"""
     )
     snapshot = page.accessibility.snapshot()
-    assert snapshot["children"][0] == {"role": "generic", "name": ""}
+    if browser_channel:
```

Comment:
this is going to break when the channels will get updated. It would be better to check by the Chromium version, see this patch which needs to get applied: https://github.com/microsoft/playwright/pull/6372

Reply:
Yeah it would break once channel is updated which would notify us that the special case for it should be removed, if we check by version then it would always pass the tests and never break. I think it is better the way since this is not the actual library code but rather tests.